fe/fe-core/src/main/java/com/starrocks/connector/hive/HiveMetaClient.java

```diff
@@ -351,14 +352,18 @@ public Map<String, List<ColumnStatisticsObj>> getPartitionColumnStats(String dbN
      * hive metastore is false. The hive metastore will throw StackOverFlow exception.
      * We solve this problem by get partitions information multiple times.
      * Each retry reduces the number of partitions fetched by half until only one partition is fetched at a time.
+     *
      * @return Hive table partitions
      * @throws StarRocksConnectorException If there is an exception with only one partition at a time when get partition,
-     * then we determine that there is a bug with the user's hive metastore.
+     *                                     then we determine that there is a bug with the user's hive metastore.
      */
     private List<Partition> getPartitionsWithRetry(String dbName, String tableName,
                                                    List<String> partNames, int retryNum) throws StarRocksConnectorException {
         int subListSize = (int) Math.pow(2, retryNum);
         int subListNum = partNames.size() / subListSize;
+        if (subListNum == 0) {
```

Comment:
why do this?

Reply:
if it's zero, the following code will throw exception

```
        List<List<String>> partNamesList = Lists.partition(partNames, subListNum);

```